pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -477,6 +479,62 @@ func (c *Controller) setupIndex() *AmbientIndexImpl {
 		}
 	})
 
+	c.configController.RegisterEventHandler(gvk.KubernetesGateway, func(oldCfg config.Config, newCfg config.Config, ev model.Event) {
+		gateway, ok := extractKubernetesGateway(newCfg)
+		if !ok {
+			return
+		}
+
+		// gateway.Status.Addresses should only be populated once the Waypoint's deployment has at least 1 ready pod, it should never be removed after going ready
+		// ignore Kubernetes Gateways which aren't waypoints
+		// TODO: should this be WaypointGatewayClass or matches a label?
+		if gateway.Spec.GatewayClassName == constants.WaypointGatewayClassName && len(gateway.Status.Addresses) > 0 {
+			scope := model.WaypointScope{Namespace: gateway.Namespace, ServiceAccount: gateway.Annotations[constants.WaypointServiceAccount]}
+
+			waypointPort := uint32(15008)
+			for _, l := range gateway.Spec.Listeners {
+				if l.Protocol == k8sbeta.ProtocolType(protocol.HBONE) {
+					waypointPort = uint32(l.Port)
+				}
+			}
+
+			// Is MustParseAddr safe? Should we use ParseAddr and handle the error?
+			ip := netip.MustParseAddr(gateway.Status.Addresses[0].Value)
+			addr := &workloadapi.GatewayAddress{
+				Destination: &workloadapi.GatewayAddress_Address{
+					Address: &workloadapi.NetworkAddress{
+						Network: c.Network(ip.String(), make(labels.Instance, 0)).String(),
+						Address: ip.AsSlice(),
+					},
+				},
+				Port: waypointPort,
+			}
+
+			updates := sets.New[model.ConfigKey]()
+			idx.mu.Lock()
+			defer idx.mu.Unlock()
+			if ev == model.EventDelete {
+				if proto.Equal(idx.waypoints[scope], addr) {
```

Comment:
if it is a delete we need to delete anyways, not just if its equal?

Reply:
huh we have that today.. probably a bug

Reply:
ff83f5993cec780dd7d36f5776f693b4363dc593 should look better